### PR TITLE
:sparkles: Proposal for a structured promotion pipeline

### PR DESCRIPTION
**Pull request by Munklinde96**

### CODE REVIEW

File: proto/rig/api/v1/project/project.proto [hunk 1]
```diff
@@ -37,3 +42,10 @@ message NotificationNotifiers {
   bool disabled = 1;
   repeated model.NotificationNotifier notifiers = 2;
 }
+
+message Pipelines {
+  // If the notifiers are disabled, notifiers from parent are not inherited even
+  // if these notifiers at this level are empty.
```
Comment (MatiasFrank) [inline proto/rig/api/v1/project/project.proto:48]: notifiers -> pipeline

File: proto/rig/model/pipeline.proto [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  repeated Phase phases = 2;
+  google.protobuf.Timestamp created_at = 3;
+  google.protobuf.Timestamp updated_at = 4;
+}
+
+message Phase {
+  // Environment ID. The project must be active in this environment.
+  string environment_id = 1;
+  // Fixed fields, that are not changed upon promotion.
+  repeated string fixed_fields = 2;
```
Comment (MatiasFrank) [inline proto/rig/model/pipeline.proto:20]: Do we use the new json path stuff here as the field identifiers?

Comment (Munklinde96) [reply proto/rig/model/pipeline.proto:20]: That is the intention

File: proto/rig/model/pipeline.proto [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+package model;
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+
+message Pipeline {
+  // Custom name for the pipeline.
+  string name = 1;
+  repeated Phase phases = 2;
```
Comment (MatiasFrank) [inline proto/rig/model/pipeline.proto:11]: What is the logic behind phase transitions? If I have two phases, one for `staging` and one for `prod`, does it then transition from `staging` to `prod`, keeping the fixed_fields defined on the `staging` or `prod` phase?

Comment (Munklinde96) [reply proto/rig/model/pipeline.proto:11]: I was thinking the following:
It computes the changes from 'prod' to 'staging', and removes changes that affect fixed_fields in 'prod'

Comment (MatiasFrank) [reply proto/rig/model/pipeline.proto:11]: Ah so the destination phase defines how to process the promotion, but the source phase defines when we transition?

**Pull request merged**